libs/feature-workspaces/src/lib/new-workspace/new-workspace.component.ts

```diff
@@ -51,139 +48,106 @@ interface SchematicCollectionForNgNew {
   templateUrl: './new-workspace.component.html',
   styleUrls: ['./new-workspace.component.scss']
 })
-export class NewWorkspaceComponent implements OnInit {
-  @ViewChildren(MatExpansionPanel)
-  matExpansionPanels: QueryList<MatExpansionPanel>;
-  schematicCollectionsForNgNew$: Observable<any>;
-  ngNewForm$ = new BehaviorSubject<FormGroup | null>(null);
-
-  selectedNode: DynamicFlatNode | null = null;
-
-  private readonly createNewWorkspace$ = new Subject<void>();
+export class NewWorkspaceComponent {
+  @ViewChild(MatVerticalStepper) verticalStepper: MatVerticalStepper;
+  commandOutput$?: Observable<IncrementalCommandOutput>;
+
+  command?: string;
+
+  ngNewForm = this.fb.group({
+    path: this.fb.control(null, Validators.required),
+    name: this.fb.control(
+      null,
+      Validators.required,
+      makeNameAvailableValidator(this.finderService)
+    ),
+    collection: this.fb.control(null, Validators.required)
+  });
+
+  schematicCollectionsForNgNew$ = this.schematicCollectionsGQL.fetch().pipe(
+    map(r => r.data.schematicCollections),
+    publishReplay(1),
+    refCount()
+  );
+
+  autofocusInput() {
+    const autofocus = (this.elementRef
+      .nativeElement as HTMLElement).querySelectorAll('.autofocus')[
+      this.verticalStepper.selectedIndex
+    ] as any;
+
+    autofocus.focus();
+  }
 
   constructor(
-    private readonly contextActionService: ContextualActionBarService,
-    private readonly matDialog: MatDialog,
+    private readonly elementRef: ElementRef,
+    private readonly router: Router,
+    private readonly dialogRef: MatDialogRef<NewWorkspaceComponent>,
+    private readonly fb: FormBuilder,
     private readonly finderService: Finder,
-    private readonly schematicCollectionsGQL: SchematicCollectionsGQL
+    private readonly schematicCollectionsGQL: SchematicCollectionsGQL,
+    private readonly workspacesService: WorkspacesService,
+    private readonly ngNewGQL: NgNewGQL,
+    private readonly commandRunner: CommandRunner
   ) {}
 
-  private static newFormGroup(finderService: Finder): FormGroup {
-    return new FormGroup({
-      path: new FormControl(null, Validators.required),
-      name: new FormControl(
-        null,
-        Validators.required,
-        makeNameAvailableValidator(finderService)
-      ),
-      collection: new FormControl(null, Validators.required)
-    });
+  handleSelection(event: MatSelectionListChange) {
+    if (event.option.selected) {
+      event.source.deselectAll();
+      event.option._setSelected(true);
```

Comment:
Can we just use `event.option.selected = true`? Seems like a semi-private method that we may not want to rely on. Are we trying to avoid change from being reported?

Reply:
This hacky code is a workaround for https://github.com/angular/material2/issues/7157.

I've left a comment to reflect that